lib/resque.rb

```diff
@@ -550,26 +550,23 @@ def sample_queues(sample_size = 1000)
   def register_hook(name, block)
     return clear_hooks(name) if block.nil?
 
-    @hooks ||= {}
-    @hooks[name] ||= []
-
     block = Array(block)
-    @hooks[name].concat(block)
+    hooks[name].concat(block)
   end
 
   # Clear all hooks given a hook name.
   def clear_hooks(name)
-    @hooks && @hooks[name] = []
+    hooks[name] = []
   end
 
-  # Retrieve all hooks
-  def hooks
-    @hooks || {}
+  # Retrieve all hooks of a given name.
+  def all_hooks_by_name(name)
+    hooks[name] || []
   end
 
-  # Retrieve all hooks of a given name.
-  def hooks(name)
-    (@hooks && @hooks[name]) || []
+  # Retrieve all hooks
+  def hooks
```

Comment:
Doesn't it break backwards compatibility to change this method from `hooks(name)` to `hooks()`?